pkg/workflow/templates.go

```diff
@@ -230,48 +142,64 @@ func generateSubchartAndAppDAGTasks(appGroup *v1alpha1.ApplicationGroup, namespa
 		return nil, err
 	}
 
-	task := v1alpha13.DAGTask{
-		Name:     utils.ConvertToDNS1123(app.Name),
+	task := *appDAGTaskHelper(app.Name, getTimeout(app.Spec.Release.Timeout), &hr)
+	task.Dependencies = func() (out []string) {
+		for _, t := range tasks {
+			out = append(out, utils.ConvertToDNS1123(t.Name))
+		}
+		return out
+	}()
+	tasks = append(tasks, task)
+
+	return tasks, nil
+}
+
+func appDAGTaskHelper(name string, timeout *v1alpha13.AnyString, hr *fluxhelmv2beta1.HelmRelease) *v1alpha13.DAGTask {
```

Comment:
Based on the usage of the return value why not just return by value instead of returning a pointer to `v1alpha13.DAGTask`?

Reply:
Done